src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -876,13 +882,20 @@ private static async Task<ConnectionMultiplexer> ConnectImplAsync(object configu
         /// </summary>
         /// <param name="configuration">The configuration options to use for this multiplexer.</param>
         /// <param name="log">The <see cref="TextWriter"/> to log to.</param>
-        public static Task<ConnectionMultiplexer> ConnectAsync(ConfigurationOptions configuration, TextWriter log = null)
+        public static async Task<ConnectionMultiplexer> ConnectAsync(ConfigurationOptions configuration, TextWriter log = null)
         {
             SocketConnection.AssertDependencies();
-            return ConnectImplAsync(configuration, log);
+
+            bool sentinel = !String.IsNullOrEmpty(configuration.ServiceName);
```

Comment:
```suggestion
            bool sentinel = !string.IsNullOrEmpty(configuration.ServiceName);
```

Reply:
Previously, a user would hit `PrepareConfig` here ultimately, which does a null check. This would cause a null ref instead - we'd need a null arg check above this access to compensate.

Reply:
Good catch.